Update documentation for junit plugins (#5259)
chef/inspec: lib/plugins/inspec-reporter-junit/README.md

# junit and junit2 reporters

This is the implementation of the junit and junit2 XML reporters.

> ## To Install These Plugins

Reviewer: ```suggestion
## Installation
```

> This plugin is included with InSpec. There is no need to install it separately.

Reviewer: ```suggestion
This plugin ships with Chef InSpec and requires no additional installation.
```

## What These Plugins Do

> `junit` is the legacy Chef InSpec JUnit reporter. It generates an XML report in Apache Ant JUnit format. The output format may be considered nonstandard in several ways. New users are advised to use `junit2`.

Reviewer: ```suggestion
`junit` is the legacy Chef InSpec JUnit reporter, which is retained for backwards compatibility. It generates an XML report in Apache Ant JUnit format. The output format is considered nonstandard in several ways. New users are advised to use `junit2`.
```

`junit2` is an updated reporter that provides JUnit output according to the schema published by [Windy Road](https://github.com/windyroad/JUnit-Schema).

## Implementation Note

> This reporter uses the REXML XML generator, but uses the more heavyweight XML library Nokogiri for testing. This is to keep packaging requirements lightweight and free of compiled dependencies.

Reviewer: ```suggestion
This reporter uses the REXML XML generator at runtime, but uses Nokogiri, a more heavyweight XML library, for testing. This design keeps packaging requirements lightweight and free of compiled dependencies.
```

Reviewer: @james-stocks "Implementation" may not be the right word choice in this case. Do you have alternative suggestions?

Author: Maybe `at runtime`? Or just leave it as `This reporter uses the REXML XML generator, but` ?

Reviewer: I went with "runtime," as it's important to differentiate between the two use types and when they're used. We can always come back to edit this later.
